extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/controlplane/store/sql/contractnegotiation/store/schema/BaseSqlDialectStatements.java

```diff
@@ -147,40 +154,17 @@ public SqlQueryStatement createNegotiationsQuery(QuerySpec querySpec) {
         return new SqlQueryStatement(sql, querySpec.getLimit(), querySpec.getOffset());
     }
 
+    @Override
+    public SqlQueryStatement createNegotiationNextNotLeaseQuery(QuerySpec querySpec) {
+        var queryTemplate = "%s LEFT JOIN %s l ON %s.%s = l.%s".formatted(getSelectNegotiationsTemplate(), leaseStatements.getLeaseTableName(), getContractNegotiationTable(), getIdColumn(), leaseStatements.getResourceIdColumn());
```

Comment:
is this left join necessary? it's overridden and it's not really filtering out leased entities